### ISSUE TITLE
background color of tab tooltip too light

### ISSUE THREAD

**Issue by OliverLew** [qutebrowser/qutebrowser#4520]
**Version info (see `:version`)**:
qutebrowser v1.5.2
Git commit: 
Backend: QtWebEngine (Chromium 69.0.3497.128)

CPython: 3.7.2
Qt: 5.12.0
PyQt: 5.11.3

sip: 4.19.13
colorama: 0.4.1
pypeg2: 2.15
jinja2: 2.10
pygments: 2.3.1
yaml: 3.13
cssutils: no
attr: 18.2.0
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebKitWidgets: yes
pdf.js: 2.0.943 (/usr/share/pdf.js/build/pdf.js)
sqlite: 3.26.0
QtNetwork SSL: OpenSSL 1.1.1a  20 Nov 2018

Style: QStyleSheetStyle
Platform: Linux-4.20.0-arch1-1-ARCH-x86_64-with-arch, 64bit
Linux distribution: Arch Linux (arch)
Frozen: False
Imported from /usr/lib/python3.7/site-packages/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib/qt/libexec, data path: /usr/share/qt

Paths:
cache: /home/oliver/.cache/qutebrowser
config: /home/oliver/.config/qutebrowser
data: /home/oliver/.local/share/qutebrowser
runtime: /run/user/1000/qutebrowser
system data: /usr/share/qutebrowser

Uptime: 1 day, 2:13:25

**Does the bug happen if you start with `--temp-basedir`?** (if applicable):
yes

**Description**
Just the background of tab tooltip is bright and light yellow, and the white foreground font is not recognizable. Can't get a screenshot of it.
However on another computer (with Manjaro) the bg color is fine (sone kind of grey).
I don't know where the problem is. Should qutebrowser handle this, or is it about Qt, or anything else?

**Comment by OliverLew** [comment]
Thank you I am using gtk2 as qt5 style, and changing to other styles can make the foreground font color black, the bg color is still yellow (it's recognizable now, though). It's still different from the tooltips of other qt programs e.g. VLC. Maybe have I messed up something else or is the tabs tooltip in qutebrowser indeed different?

**Comment by jgkamat** [comment]
I get light yellow bg and black fg. I think (as The-Compiler said before) that this is coming from your qt style. If you want to try another program, could you try kde falkon, their tabs have the same tooltip popup for me.

**Comment by OliverLew** [comment]
Yes I agree it should be because my qt settings. but the tooltip in qutebrowser is different from any other qt program I have like vlc (I also tried falkon, too), the latter all give a tooltip of white fg and grey bg just like the screenshot above. Only qutebrowser has yellow bg and black(or white if I use gtk2 style) fg.
I can live with it though just wonder if the tooltips are different (unlikely?) or the style is overridden somewhere.

**Comment by The-Compiler** [comment]
Okay, that's odd. Reopening it for now, though I don't really know how I could reproduce the issue.

**Comment by OliverLew** [comment]
OK, maybe I didn't set the environment variable QT_QPA_PLATFORMTHEME right. I was using `qt5ct`, but after changing to `gtk2` according to [Arch Wiki](https://wiki.archlinux.org/index.php?title=Uniform_look_for_Qt_and_GTK_applications), everything is good now!

I don't know why the old setting will make qutebrowser behave differently from other qt programs though. If this doesn't count as a bug, maybe this issue can be closed?

**Comment by The-Compiler** [comment]
I can now reproduce this (even with tooltips on the page), but not in KDE Falkon - so this certainly needs some more investigation.

**Comment by jjzmajic** [comment]
Just a pinging to say I have the same issue with or without setting the QT env variable. Must be something else but I cannot figure out what.

**Comment by HiTuX** [comment]
I occasionaly see this problem and it affects tab tooltip and and on page tooltip only when **black theme** is used. I remember it was resolved by some updates ! now I see it again!
I found that setting `qt5ct` to any style other than `gtk2` make the tooltip _black on yallow_, very clear but did fellow the black theme colors. Also the problem is with `Default` palette, change to any other costum palette works as expected.
Not sure if this could be a solution but one can ckeck `tooltip-simple.qss` under `Style sheets` tab in `qt5ct`. Looking good for me ;)

`tooltip-simple.qss` contains the following lines:
```css
QToolTip{
  background: QLinearGradient(x1: 0, y1: 0, x2: 1, y2: 1, stop: 0 palette(window), stop: 1 palette(alternate-window));
  border-radius: 3px;
  border: 1px solid palette(highlight);
  padding: 1px;
  color: palette(text);
}
```
Regarding `Falkon`, I think it does NOT use `qt5ct custom style` ! read from terminal:
```
qt5ct: custom style sheet is disabled
```
May be qutebrowser should use somthing similar to avoid such a problem.
Hoping this could help.

**Comment by jjzmajic** [comment]
.

**Comment by The-Compiler** [comment]
@jjzmajic huh?

**Comment by jjzmajic** [comment]
Oh dammit, I left the tab open while searching for the same issue. Must have been an accidental click. Sorry about that.

Since I'm posting already, any progress on the issue?

**Comment by The-Compiler** [comment]
Nothing I'm aware of.

**Comment by Schubisu** [comment]
@HiTuX's suggestion worked well for me. I'm also using gtk2 theme in qt5ct, it's been sufficient to set the colorscheme from default to any of the custom schemes.
Since a recent update, my tooltips had a bright yellow background with white foreground color, setting the color scheme to `simple` (which looks very similar to default) gives me yellow bg with black fg, like it's been before. Only the default scheme renders them unreadable.

**Comment by JonnyHaystack** [comment]
I had this problem using Arc Dark GTK theme and `QT_QPA_PLATFORMTHEME=gtk2`.

I worked around it by doing the following:
1. Set `QT_QPA_PLATFORMTHEME=qt5ct` instead
2. In qt5ct, set style: gtk2, standard dialogs: gtk2, palette: default
3. Change to Style Sheets tab, and create a new file (I called it tooltip-gtk2.qss but it shouldn't matter)
4. Put the following contents inside:
```
QToolTip{
  background: QLinearGradient(x1: 0, y1: 0, x2: 0, y2: 0, stop: 0 palette(window), stop: 1 palette(alternate-window));
  border-radius: 3px;
  border: 1px solid #000000;
  padding: 1px;
  color: palette(text);
}
```
5. Click Save then Ok
6. Make sure to check the box next to this new file so that it will be applied to the theme
7. Click Apply

Basically I took tooltip-simple.qss and removed the gradient and changed the border to black, so it looks just like the tooltips are supposed to look in this theme. This may not be the cleanest way to do it but I know nothing about Qt themeing.

**Comment by C0DK** [comment]
Even though we can find hacks around it, it would be great if this was solved on in the qutebrowser source code. As it seems it is only this program that has this weird bug - so finding out exactly why, would be great! But i get that i might be hard. I'm trying to dive into the source code.

**Comment by The-Compiler** [comment]
@C0DK Sure. I guess nobody will argue with that! However, at least I have no idea where to start or what would cause this.

**Comment by tamis-laan** [comment]
I'd like to note I have the same problem, using the environment variables (shell: fish):
```
set -gx QT_STYLE_OVERRIDE gtk2
set -gx QT_QPA_PLATFORMTHEME gtk2
```
No expert on the gtk and qt but maybe helpful.

**Comment by Barbarossa93** [comment]
Actually, you don't need gtk2 to adjust the tooltips.  Since my gtk theme doesn't have a gtk2 variant, I went with these steps:

0. Indeed, set `QT_QPA_PLATFORMTHEME=qt5ct` as an environment variable
1. Use whatever settings you want for the appearance tab in `qt5ct`
2. Add the stylesheet that @JonnyHaystack mentioned
3. You don't have to use those exact values. For example, I use:
```
QToolTip{
	background: #f8f0dc;
	border-radius: 0px;
	border: 0px solid #000000;
	padding: 2px;
	color: #5c6a72;
}
```
4. Follow the rest of @JonnyHaystack directions

This of course will only affect the tooltips. AFAIK, there aren't other directly dependent qt UI elements in qutebrowser, but then again, I've only been using qutebrowser for a few days so I probably missed something :)

**Comment by The-Compiler** [comment]
I suspect this could be due to a similar cause as https://github.com/qutebrowser/qutebrowser/issues/6336#issuecomment-1106958498
